src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  Icon,
+  Input,
+  Modal,
+  Notification,
+} from "@canonical/react-components";
+import { useFormik } from "formik";
+import { createFineGrainedTlsIdentity } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+  const [error, setError] = useState(false);
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoading(true);
+
+      createFineGrainedTlsIdentity(values.clientName)
+        .then((tokenPreResponse) => {
+          const tempToken = base64EncodeTokenDetails(tokenPreResponse);
+
+          setIsLoading(false);
+          setToken(tempToken);
+
+          void queryClient.invalidateQueries({
+            queryKey: [queryKeys.identities],
+          });
+        })
+        .catch(() => {
+          setIsLoading(false);
+          setError(true);
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={onClose}
+      className="create-tls-identity"
+      title="Generate trust token"
+      buttonRow={
+        <>
+          {token && (
+            <Button
+              className="u-no-margin--bottom"
+              onClick={() => {
+                navigator.clipboard.writeText(token);
+              }}
+              type="button"
+              hasIcon
+            >
+              <Icon name="copy" />
+            </Button>
+          )}
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            onClick={token ? onClose : () => void formik.submitForm()}
+            disabled={formik.values.clientName.length === 0 || isLoading}
+            loading={isLoading}
+          >
+            {token ? `Close` : `Generate Token`}
+          </ActionButton>
+        </>
+      }
+    >
+      {error && (
+        <Notification
+          severity="negative"
+          title="Fine grained identity token creation failed"
+        ></Notification>
+      )}
+      {!token && (
+        <Input
+          id="clientName"
+          type="text"
+          label="Client Name"
+          onBlur={formik.handleBlur}
+          onChange={formik.handleChange}
+          value={formik.values.clientName}
+        />
```

Comment:
We are creating an identity, this should be called "Identity name", where does the "client name" idea come from?

Reply:
This is the terminology that the API and documentation uses. Yes, when the identity surfaces, it appears as the "Identity name", but I believe there is more to it than just the name and that staying true to the API may avoid potential issues in the future if the API changes against our mental model of "Identity name".

Reply:
Right good points. I see why the docs are doing it. They are tailoring towards a new command line client. I think in the UI, the target audience is a bit different. A user might create the token for a new client, but it could also be for another person (think an admin creating the token for a colleague). In the API layer we use, this is just "name". I agree, a "Name" label would be confusing here when creating a token. The entity that is created is referred to as "Identity" in the UI. So maybe "Identity name" is the better description to cover all target use cases. Wdyt?

Reply:
Sure, I think for the sake of the end user we can change client name to identity name. I do think that "Client name" allows for the user to think about their nming decision a little more, whilst "name" just implies a casual identity. To me, it's like the difference between "James's token" and "token 1" or "Chrome browser" and "laptop", because "client" implies the user should think more about how, why and where they are using it, as opposed to just what to call it. Especially as there is no renaming function and poor naming can make for confusing identities. Thoughts?

Reply:
I think we should be consistent with the identity name as that is how the entity is presented after creation.

Reply:
Alrighty! "Identity name" it is.